common/src/main/java/io/netty/util/internal/UnpaddedInternalThreadLocalMap.java

```diff
@@ -32,7 +32,7 @@
  */
 class UnpaddedInternalThreadLocalMap {
 
-    static ThreadLocal<InternalThreadLocalMap> slowThreadLocalMap;
+    static final ThreadLocal<InternalThreadLocalMap> slowThreadLocalMap = new ThreadLocal<InternalThreadLocalMap>();
```

Comment:
@windie interesting. I actually checked the assembly and it seems legit. When adding the final keyword, the `slowThreadLocalMap` is "inlined" into the jitted code and you avoid one load + a typecheck.

However, I wouldn't expect this to make a difference in real code. Still nice, as it seems to be one of the few examples where final affects code generation.